### PR TITLE
Update OSSpecofic plugin code and build script not to use reflection calls

### PR DESCRIPTION
**Pull request by makiam**
Use orangeextensions lib at compile time to allow cross compile mac specific code

### CODE REVIEW

File: OSSpecific.xml [hunk 1]
```diff
@@ -3,9 +3,10 @@
 <project name="OSSpecific" default="dist" basedir=".">
 
   <!-- set global properties for this build -->
-  <property name="src" value="OSSpecific/src" />
-  <property name="build" value="OSSpecific/build" />
-  <property name="docs" value="OSSpecific/docs" />
+  <property name="src" value="${ant.project.name}/src" />
```
Comment (Sailsman63) [inline OSSpecific.xml]: These changes do not belong in this PR. If you would like to make some style changes to the build scripts as a whole, please raise an issue for discussion. This has nothing to do with removing the reflection code from `OSSpecific`.

File: OSSpecific/src/artofillusion/osspecific/MacOSPlugin.java [hunk 2]
```diff
@@ -15,16 +15,21 @@ Changes copyright (C) 2017 by Maksim Khramov
 import artofillusion.ui.*;
 import buoy.event.*;
 import buoy.widget.*;
+import com.apple.eawt.Application;
+import com.apple.eawt.ApplicationEvent;
 import java.awt.*;
 import java.io.*;
-import java.lang.reflect.*;
+import java.util.logging.Level;
```
Comment (Sailsman63) [inline OSSpecific/src/artofillusion/osspecific/MacOSPlugin.java]: We have not decided on a logging strategy. (#50)

Please refrain from adding logging code that is specific to a certain logger configuration, especially as the core concern here has nothing to do with logging.

**Comment by Sailsman63** [comment]
@peastman: When you get a chance, can you test this branch? IIUC, newer OSX systems that have Oracle Java 7+, but not AppleJava, will not have the extension classes available. I would expect, though, that this would mean that the reflective version of the code would not work either...

Sorry, don't have a mac to test on.

**Comment by peastman** [comment]
Like @Sailsman63 I'm confused about what this change is for.  What problem is it trying to solve?

**Comment by makiam** [comment]
This update to more clean, easy to understand and update code. For future... Now only Quit and Preferences menu are moved to mac-specific menu. All other menu items also may be moved to it

**Comment by makiam** [comment]
I compile and run this with JDK 8 update 131 on my Mac. As expected Preferences and Quit menu items moved to mac menu

**Comment by Sailsman63** [comment]
>I compile and run this with JDK 8 update 131 on my Mac.

Okay, but does you Mac also have leftover bits from an AppleJava installation? What we need tested is whether it works on a Modern mac, with _no traces_ of AppleJava code.

>As expected Preferences and Quit menu items moved to mac menu.

Does this mean that they are _not_ moved to the mac menu in the current master code?

**Comment by makiam** [comment]
> Okay, but does you Mac also have leftover bits from an AppleJava installation

I have fresh elCapitan 10.11.6 installation on it. There are no Apple java by default

> Does this mean that they are not moved to the mac menu in the current master code?

Means no regression in UI

**Comment by Sailsman63** [comment]
@peastman Hate to bug you about this one again, but can you clarify something for me?

- Does the current `master` version of `OSSpecific` have UI issues, such as some standard menu items not appearing where a mac user would expect them?
- If so, does this change fix that?

**Comment by peastman** [comment]
I spent a little while testing the master branch, and didn't find any obvious problems.  At any rate, the most obvious Mac-specific things seemed to be correct.  The menu bar is at the top of the screen, the Preferences and Quit menu items are both in the application menu, and it shows the marker when the scene is modified.

**Comment by Sailsman63** [comment]
@peastman Okay, thanks. In that case, I'm dropping this one. While I sort of sympathize with the intent to reduce reflection, I don't think that its worth another dependency in this case.

(The other option for dropping the reflection code would be to compile `OSSpecific` on mac only. This would be doable through travis.)

**Comment by peastman** [comment]
That sounds good to me.
